pkg/infrastructure/aws/clusterapi/ignition.go

```diff
@@ -0,0 +1,71 @@
+package clusterapi
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/ec2"
+	"github.com/sirupsen/logrus"
+	capa "sigs.k8s.io/cluster-api-provider-aws/v2/api/v1beta2"
+	k8sClient "sigs.k8s.io/controller-runtime/pkg/client"
+
+	"github.com/openshift/installer/pkg/asset/manifests/capiutils"
+	"github.com/openshift/installer/pkg/infrastructure/clusterapi"
+	awstypes "github.com/openshift/installer/pkg/types/aws"
+	"github.com/openshift/installer/pkg/types/dns"
+)
+
+func editIgnition(ctx context.Context, in clusterapi.IgnitionInput) ([]byte, error) {
+	if in.InstallConfig.Config.AWS.UserProvisionedDNS != dns.UserProvisionedDNSEnabled {
+		return in.BootstrapIgnData, nil
+	}
+
+	awsCluster := &capa.AWSCluster{}
+	key := k8sClient.ObjectKey{
+		Name:      in.InfraID,
+		Namespace: capiutils.Namespace,
+	}
+	if err := in.Client.Get(ctx, key, awsCluster); err != nil {
+		return nil, fmt.Errorf("failed to get AWSCluster: %w", err)
+	}
+
+	awsSession, err := in.InstallConfig.AWS.Session(ctx)
+	if err != nil {
+		return nil, fmt.Errorf("failed to get aws session: %w", err)
+	}
+
+	securityGroupIDs := []*string{}
```

Comment:
We can preallocate the slice:
```suggestion
	securityGroupIDs := make([]*string, 0, len(awsCluster.Status.Network.SecurityGroups))
```